src/ra_wm_handler.h

```diff
@@ -0,0 +1,132 @@
+#ifndef RA_WM_HANDLER_H
+#define RA_WM_HANDLER_H
+#pragma once
+
+#include <WTypes.h>
+#include <yvals.h>
+
+#ifndef _NORETURN
+#define _NORETURN
+#endif // !_NORETURN
+
+// References:
+// https://docs.microsoft.com/en-us/windows/desktop/api/Winuser/nf-winuser-defdlgprocw
+
+namespace ra {
+namespace ui {
+
+// We can make this an abstract interface later
+/// <summary>This class serves as a base Windows Message handler.</summary>
+/// <remarks>
+///   Only Messages that are guaranteed to be processed or have the same behavior
+///   will be here. Any additional handlers should derive this class
+/// </remarks>
+struct WM_Handler
+{
+#pragma region Constructors
+    WM_Handler() noexcept = default;
+    virtual ~WM_Handler() noexcept = default;
+    WM_Handler(const WM_Handler&) = delete;
+    WM_Handler& operator=(const WM_Handler&) = delete;
+    inline constexpr WM_Handler(WM_Handler&&) noexcept = default;
+    inline constexpr WM_Handler& operator=(WM_Handler&&) noexcept = default;
+#pragma endregion
+
+    // Basic rule: Messages with pointers (the first hwnd is not part of it) are
+    //             sent, otherwise they are posted. If you don't want it to processed immediate post it the message queue, but make sure it has no pointers.
+#pragma region Event Handlers
+    // This base method should only be used in the dialogs message queue and not it's procedure 
+    _NODISCARD virtual BOOL OnInitDialog(_In_ HWND hwnd, _In_ HWND hwndFocus, _In_ LPARAM lParam) noexcept;
+    _NORETURN virtual void OnCommand(_In_ HWND hwnd, _In_ int id, _In_ HWND hwndCtl, _In_ UINT codeNotify) noexcept;
```

Comment:
All of these could be const qualified but didn't want to force it, just in case someone wants to make a derived class stateful.